src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1189,19 +1189,30 @@ public Component getTableCellRendererComponent(JTable table, Object value,
                 Icon icon = chooser.getIcon(file);
                 setIcon(icon);
 
-            } else if (value instanceof Long) {
-                long len = ((Long) value) / 1024L;
+            } else if (value instanceof Long len) {
```

Comment:
I think this code needs to be extracted into a function, and a unit test added.  Testing with different sizes and supported locales - let's say en_IN (Asian group separator) and ja_JP or zh_CN (no plural forms).

Reply:
JDK does not do unit tests in this way. There is no place to put it.
We do functional "system" level regression tests.

Reply:
I guess extracting this code to a function won't give much advantage as it is not called from multiple places.

Reply:
I do not understand how it is possible to develop without unit tests.  

I would test this function with 0, 1, and (N-1, N, N+1) where N is in range of
1000, 1024, 1_000_000, 1024*1024, 1_000_000_000, 1024^3, 1_000_000_000_000, 1024^4.
(I doubt you can create 1TB test file)